tests/test_node.cpp

```diff
@@ -51,16 +51,16 @@ BOOST_FIXTURE_TEST_CASE(test_node_validate_id, FixtureInitClose)
 {
     int index;
 
-    error = EN_addnode(ph, (char *)"N2", EN_JUNCTION);
+    error = EN_addnode(ph, (char *)"N2", EN_JUNCTION, &index);
     BOOST_REQUIRE(error == 0);
 
-    error = EN_addnode(ph, (char *)"N 3", EN_JUNCTION);
+    error = EN_addnode(ph, (char *)"N 3", EN_JUNCTION, &index);
     BOOST_REQUIRE(error == 252);
 
-    error = EN_addnode(ph, (char *)"N\"3", EN_JUNCTION);
+    error = EN_addnode(ph, (char *)"N\"3", EN_JUNCTION, &index);
```

Comment:
Note: This is ok with EPANET UI.
![image](https://user-images.githubusercontent.com/2945956/56365785-59556e00-61fa-11e9-86c1-ec59a0a710c6.png)

Reply:
This observation about the UI accepting `N\"3` as a valid ID raises a number of points:

1. Not only will the UI accept the ID but so will the `EN_open` API function that reads in an input file.
2. The new update to the UI will not allow a space, semi-colon, or double quote to be entered into an ID field in the Property Editor (which is a good thing).
3. The engine code's input file reader does not do any checking of ID names. If they exceed 31 characters they are truncated. If the ID in the original input line had a pair of double quotes for the ID name field, the name will be truncated at the first quote and the characters within the quotes will be treated as a separate token. This behavior may (or may not) trigger an input error.

It's my opionion that we stick with the current restriction on valid ID names used with the EN_add functions but also extend it to the names processed through an .INP file, even if this might affect backward compatibility.